File: algorithms/selectionSort/SelectionSort.java
class SelectionSort {
	public static void main(String[] args) {
		byte[] arr = new byte[] { 64, 25, 12, 22, 11 };
		selectionSort(arr);
		for (byte b : arr) {
			System.out.println(b);
		}
	}

	private static void selectionSort(byte[] arr) {
		for (int i = 0; i < arr.length - 1; i++) {
			int smallestIndex = i;
			for (int j = i + 1; j < arr.length; j++) {
				if (arr[smallestIndex] > arr[j])
					smallestIndex = j;
			}

			if (i != smallestIndex) {
				arr[i] ^= arr[smallestIndex];
				arr[smallestIndex] ^= arr[i];
				arr[i] ^= arr[smallestIndex];
			}
		}
	}
}

File: python/algorithms/sort/selection_sort.py
def selectionSort(nums):
    for i in range(0, len(nums)):
        smallestIndex = i
        for j in range(i + 1, len(nums)):
            if nums[j] < nums[smallestIndex]:
                smallestIndex = j
        
        # nums[i], nums[smallestIndex] = nums[smallestIndex], nums[i] # tuple swap
        if i != smallestIndex:
            nums[i] ^= nums[smallestIndex]
            nums[smallestIndex] ^= nums[i]
            nums[i] ^= nums[smallestIndex]

    return nums

nums = [2, 8, 5, 3, 9, 4, 1]
print(selectionSort(nums))